### PR TITLE
Add WarningCollector to MetastoreContext

### PR DESCRIPTION
**Pull request by NikhilCollooru**
## Description
Add WarningCollector to MetastoreContext

## Motivation and Context
Goal is to add a warning based on metastore interactions.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by abhiseksaikia — APPROVED** [review]
LGTM!

File: presto-spi/src/main/java/com/facebook/presto/spi/security/AccessControlContext.java [hunk 1]
```diff
@@ -24,12 +25,14 @@ public class AccessControlContext
     private final QueryId queryId;
     private final Optional<String> clientInfo;
     private final Optional<String> source;
+    private final WarningCollector warningCollector;
 
-    public AccessControlContext(QueryId queryId, Optional<String> clientInfo, Optional<String> source)
+    public AccessControlContext(QueryId queryId, Optional<String> clientInfo, Optional<String> source, WarningCollector warningCollector)
```
Comment (ajaygeorge) [inline presto-spi/src/main/java/com/facebook/presto/spi/security/AccessControlContext.java:30]: curious why AccessControlContext needs the warningCollector as well.

**Review by ajaygeorge — APPROVED** [review]
LGTM % a question

File: presto-spi/src/main/java/com/facebook/presto/spi/security/AccessControlContext.java (hunk 1 above)
Comment (NikhilCollooru) [reply presto-spi/src/main/java/com/facebook/presto/spi/security/AccessControlContext.java:30]: SqlStandardAccessControl methods create MetastoreContext.. and AccessControlContext is the way to pipe the warningCollector till there.

**Pull request merged**